src/json/serialization.cr

```diff
@@ -393,7 +393,7 @@ module JSON
       def self.new(pull : ::JSON::PullParser)
         location = pull.location
 
-        discriminator_value = nil
+        discriminator_value : String | Int64 | Bool | Nil = nil
```

Comment:
I guess we don't need Float64 here, but it seems the atomic values of JSON::Any are wanted here, except for Float64

Reply:
This variable really doesn't need any type restrictions at all.